site/soc/software/charts/osh/openstack-ceph-config/ceph-config.yaml

```diff
@@ -53,4 +53,7 @@ data:
       job_namespace_client_key: false
       storageclass_cephfs: false
       storageclass_rbd: false
+    pod:
+      resources:
+        enabled: {{ openstack_helm_pod_resources_enabled['ceph'] }}
```

Comment:
I fail to see how just enabling pod resources seem to be working. I was expecting something requesting specific [CPU/Memory/Disk](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#how-pods-with-resource-requests-are-scheduled). What am I missing here?

Reply:
Yeah so...where to start :D


Airship uses the openstack-helm charts to deploy openstack rigth? So those charts already have limits/requests in place, but they need to be activated by setting the chart value `pod.resources.enabled` to `true`, otherwise it wont apply them on chart generation.

Airship uses this files to pass values to the osh charts, thus having a place for easy overriding. Adding this values here means that they will be applied on the osh charts when building and deploying them. See here[0] for where the limits/requests are in the upstream chart and here[1] for how they are applied depending on the value of `pod.resources.enabled` and here[2] the actual function from htk that expands those values into the k8s yaml

[0] https://github.com/openstack/openstack-helm-infra/blob/master/ceph-mon/values.yaml#L102
[1] https://github.com/openstack/openstack-helm-infra/blob/master/ceph-mon/templates/daemonset-mon.yaml#L123
[2] https://github.com/openstack/openstack-helm-infra/blob/master/helm-toolkit/templates/snippets/_kubernetes_resources.tpl

Reply:
thanks, much clearer now.

Reply:
> thanks, much clearer now.

are you sure? re-reading makes it sound less clearer to me :rofl: